RELEASE.md

```diff
@@ -172,6 +172,89 @@ Steps:
 
 ---
 
+## 🍎 LLVM Apple Containers Release Process
+
+> **📋 Status**: Infrequent releases (approximately every 3 years)
+> **Use Case**: Cross-compilation for macOS targets from Linux environments
+
+### When to Release LLVM Containers
+- New Xcode/macOS SDK versions are available
```

Comment:
Maybe also worth mentioning: when a Go version upgrade requires a new version of the MacOS SDK, e.g. https://tip.golang.org/doc/go1.25#darwin?

Reply:
Do you mind contributing to this PR and including those relevant notes?